packages/user_packages/iotcc_mqtt_dev.py

```diff
@@ -0,0 +1,186 @@
+        :param registry: the instance of ResourceRegistryPerPackage of the package
+        :return:
+        """
+        from liota.entities.devices.simulated_device import SimulatedDevice
+        from liota.entities.metrics.metric import Metric
+        import copy
+
+        # Acquire resources from registry
+        self.iotcc = registry.get("iotcc_mqtt")
+        # Creating a copy of edge_system object to keep original object "clean"
+        self.iotcc_edge_system = copy.copy(registry.get("iotcc_mqtt_edge_system"))
+
+        self.reg_devices = []
+        num_devices = 5
+
+        for i in range(0, num_devices):
+            try:
+                # Register device
+                device = SimulatedDevice(socket.gethostname() + "-ChildDev" + str(i))
+                log.info("Registration Started for Device {0}".format(device.name))
+                # Device Registration attempts
+                reg_attempts = 0
+                # Started Device Registration attempts
+                while reg_attempts <= retry_attempts:
+                    try:
+                        reg_dev = self.iotcc.register(device)
+                        break
+                    except Exception, e:
+                        if reg_attempts == retry_attempts:
+                            raise
+                        reg_attempts += 1
+                        log.error(
+                            '{0} Device Registration Failed with following error - {1}'.format(device.name, str(e)))
+                        log.info('{0} Device Registration: Attempt: {1}'.format(device.name, str(reg_attempts)))
+                        time.sleep(delay_retries)
+
+                self.reg_devices.append(reg_dev)
+                # Attempts to set device relationship with edge system
+                relationship_attempts = 0
+                while relationship_attempts <= retry_attempts:
+                    try:
+                        self.iotcc.create_relationship(self.iotcc_edge_system, reg_dev)
+                        break
+                    except Exception, e:
+                        if relationship_attempts == retry_attempts:
+                            raise
+                        relationship_attempts += 1
+                        log.error(
+                            '{0} Device relationship with Edge System failed with following error - {1}'.format(
+                                device.name, str(e)))
+                        log.info(
+                            '{0} Device Relationship: Attempt: {1}'.format(device.name, str(relationship_attempts)))
+                        time.sleep(delay_retries)
+                # Use the device name as identifier in the registry to easily refer the device in other packages
+                device_registry_name = socket.gethostname() + "-ChildDev" + str(i)
+                registry.register(device_registry_name, reg_dev)
+
+                # Attempts to set properties for registered device
```

Comment:
Probably comment saying that it is trying to set multiple child properties by passing in a dictionary object would be more clear.

Reply:
Added